src/ONYXKEYS.js

```diff
@@ -76,6 +79,9 @@ export default {
     // Contains the users's block expiration (if they have one)
     NVP_BLOCKED_FROM_CONCIERGE: 'private_blockedFromConcierge',
 
+    // Does this user have push notifications enabled?
+    PUSH_NOTIFICATIONS_ENABLED: 'pushNotificationsEnabled',
```

Comment:
Am I mistaken, or aren't we using deviceID instead or userID here?

Reply:
> // Does this **device** have push notifications enabled?

Reply:
Well, yeah it's an NVP so it is tied to a user. So this should say:

> // Does this user have push notifications enabled **on this device**?

Reply:
It's maybe also worth noting that this isn't actually the same as the NVP. Let me explain:

- In the back-end, the NVP looks like:

    ```json
    {
        [deviceID]: [
            {isEnabled: true, timestamp: 123456},
            {isEnabled: false, timestamp: 7890123}
        ],
        [deviceID2]: [
            ...
            ...
        ],
        ...
    }
    ```
    
- At one point, I wrote this PR such that the front-end was aware of and consumed that structure directly. But then I realized a few things:
    - The deviceID will _never_ change while the same javascript context is running. So only the value for the one deviceID even matters
    - Because front-end is the source of truth for whether or not the user is opted-in to push notifications, it never actually needs to read the value of that NVP.
    - The code was unnecessarily complicated because we needed access to the deviceID whenever we were using `Onyx.connect` to subscribe to this key in order to find the data we needed. This led to circular dependencies which I was able to resolve by creating sub-modules that otherwise weren't needed.
    
- So ultimately I decided to rip that up and the result is much simpler:
    - `ONYXKEYS.PUSH_NOTIFICATIONS_ENABLED` is just a boolean from the front-end's perspective, so consuming it is simple and straightforward. It also gets cleared on logout and never needs to be fetched over a network.
    - We only need the deviceID when we make the `OptInToPushNotifications`/`OptOutOfPushNotifications` API calls.